File: MFDLabs.Grid.Bot/Commands/HelloWorld.cs
using System.Threading.Tasks;
using Discord.WebSocket;
using MFDLabs.Grid.Bot.Extensions;
using MFDLabs.Grid.Bot.Interfaces;
using MFDLabs.Grid.Bot.Utility;

namespace MFDLabs.Grid.Bot.Commands
{
    internal class HelloWorld : IStateSpecificCommandHandler
    {
        public string CommandName => "Grid Server Hello World";
        public string CommandDescription => "Trys to invoke a HelloWorld SOAP request to a Grid Server instance via SoapUtility.";
        public string[] CommandAliases => new string[] { "hw", "helloworld" };
        public bool Internal => false;
        public bool IsEnabled { get; set; } = true;

        public async Task Invoke(string[] messageContentArray, SocketMessage message, string originalCommand)
        {
            await message.ReplyAsync(await SoapUtility.Singleton.HelloWorldAsync());
        }
    }
}

File: Source/Lib/Commands/HelloWorld.ts
import { Message } from 'discord.js';
import { SOAPUtil } from '../RCCService/SOAPUtility';
import { StateSpecificCommandHandler } from '../CommandUtility/Types/StateSpecificCommandHandler';
import { AdminUtil } from '../Administration/AdminUtil';

class HelloWorld implements StateSpecificCommandHandler {
	public CommandName = 'Hello World';
	public CommandDescription = 'Invokes a Hello World request.';
	public Internal = true;

	public Command = ['hw', 'helloworld'];
	public Callback = async (_messageContent: string[], messageRoot: Message) => {
		if (!AdminUtil.RejectIfNotAdmin(messageRoot)) return;

		messageRoot.reply(await SOAPUtil.HelloWorld());
	};
}

export = new HelloWorld();
